src/__tests__/extensions/toolbar.test.ts

```diff
@@ -0,0 +1,199 @@
+            set_config: jest.fn(),
+        } as unknown as PostHog
+        toolbar = new Toolbar(instance)
+    })
+
+    beforeEach(() => {
+        ;(window as any).ph_load_toolbar = jest.fn()
+        delete (window as any)['_postHogToolbarLoaded']
+    })
+
+    describe('maybeLoadToolbar', () => {
+        const localStorage = {
+            getItem: jest.fn(),
+            setItem: jest.fn(),
+        }
+        const history = { replaceState: jest.fn() }
+
+        const defaultHashState = {
+            action: 'ph_authorize',
+            desiredHash: '#myhash',
+            projectId: 3,
+            projectOwnerId: 722725,
+            readOnly: false,
+            token: 'test_token',
+            userFlags: {
+                flag_1: 0,
+                flag_2: 1,
+            },
+            userId: 12345,
+        }
+
+        const withHashParamsFrom = (
+            hashState: Record<string, any> | string = defaultHashState,
+            key: string = 'state'
+        ) => ({
+            access_token: 'access token',
+            [key]: encodeURIComponent(_isString(hashState) ? hashState : JSON.stringify(hashState)),
+            expires_in: 3600,
+        })
+
+        const withHash = (hashParams: Record<string, any>) => {
+            return Object.keys(hashParams)
+                .map((k) => `${k}=${hashParams[k]}`)
+                .join('&')
+        }
+
+        const aLocation = (hash?: string) => {
+            if (_isUndefined(hash)) {
+                hash = withHash(withHashParamsFrom())
+            }
+
+            return {
+                hash: `#${hash}`,
+                pathname: 'pathname',
+                search: '?search',
+            }
+        }
+
+        beforeEach(() => {
+            localStorage.getItem.mockImplementation(() => {})
+
+            jest.spyOn(toolbar, 'loadToolbar')
+        })
+
+        it('should initialize the toolbar when the hash state contains action "ph_authorize"', () => {
+            // the default hash state in the test setup contains the action "ph_authorize"
+            toolbar.maybeLoadToolbar(aLocation(), localStorage, history)
+
+            expect(toolbar.loadToolbar).toHaveBeenCalledWith({
+                ...toolbarParams,
+                ...defaultHashState,
+                source: 'url',
+            })
+        })
+
+        it('should initialize the toolbar when there are editor params in the session', () => {
+            // if the hash state does not contain ph_authorize then look in storage
+            localStorage.getItem.mockImplementation(() => JSON.stringify(toolbarParams))
+
+            const hashState = { ...defaultHashState, action: undefined }
+            toolbar.maybeLoadToolbar(aLocation(withHash(withHashParamsFrom(hashState))), localStorage, history)
+
+            expect(toolbar.loadToolbar).toHaveBeenCalledWith({
+                ...toolbarParams,
+                source: 'localstorage',
+            })
```

Comment:
this test changed as well as being converted. it wasn't actually testing what it claimed it was, it's still maybe too confusing